packages/tasks_panel/src/commands/edit-task.ts

```diff
@@ -1,41 +1,28 @@
-import { window } from "vscode";
+import { commands, window } from "vscode";
 import { ConfiguredTask } from "@sap_oss/task_contrib_types";
-import { TaskTreeItem } from "../view/task-tree-item";
 import { messages } from "../i18n/messages";
 import { getSWA } from "../utils/swa";
-import { createTaskEditorPanel, getTaskInProcess } from "../panels/panels-handler";
-import { ITasksProvider } from "../services/definitions";
-import { find, has, isMatch } from "lodash";
-import { serializeTask } from "../utils/task-serializer";
-import { getLogger } from "../logger/logger-wrapper";
+import { find, isMatch } from "lodash";
+import { createTaskEditorPanel, getTaskEditorPanel, getTaskInProcess } from "../panels/panels-handler";
+import { ITasksProvider } from "../../src/services/definitions";
+import { getLogger } from "../../src/logger/logger-wrapper";
 
 export async function editTreeItemTask(
-  tasksProvider: ITasksProvider,
+  taskProvider: ITasksProvider,
   readResource: (file: string) => Promise<string>,
-  treeItem: TaskTreeItem
+  task: ConfiguredTask
```

Comment:
rename to `targetTask` ?

E.g your target is named `task`
You matching is called `_task`

Not the most distinct names